browser/net/brave_reduce_language_network_delegate_helper.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+#include <vector>
+
+#include "brave/browser/net/brave_reduce_language_network_delegate_helper.h"
+
+#include "base/strings/string_split.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/components/brave_shields/browser/brave_farbling_service.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "brave/components/brave_shields/common/features.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/language/core/browser/language_prefs.h"
+#include "components/language/core/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "net/base/net_errors.h"
+
+using brave_shields::ControlType;
+
+namespace brave {
+
+namespace {
+constexpr char kAcceptLanguageMax[] = "en-US,en;q=0.9";
+}
+
+std::string FarbleAcceptLanguageHeader(
+    const GURL& tab_origin,
+    Profile* profile,
+    PrefService* pref_service,
+    HostContentSettingsMap* content_settings) {
+  std::string languages =
+      pref_service->Get(language::prefs::kAcceptLanguages)->GetString();
+  std::string accept_language_string = language::GetFirstLanguage(languages);
+  // If the first language is a multi-part code like "en-US" or "zh-HK",
+  // extract and append the base language code to |accept_language_string|.
+  const std::vector<std::string> tokens = base::SplitString(
+      accept_language_string, "-", base::TRIM_WHITESPACE, base::SPLIT_WANT_ALL);
+  if (!tokens.empty() && tokens[0] != accept_language_string) {
+    accept_language_string += "," + tokens[0];
+  }
+  // Add a fake q value after the language code.
+  std::vector<std::string> q_values = {";q=0.5", ";q=0.6", ";q=0.7", ";q=0.8",
+                                       ";q=0.9"};
+  brave::FarblingPRNG prng;
+  if (g_brave_browser_process->brave_farbling_service()
+          ->MakePseudoRandomGeneratorForURL(
+              tab_origin, profile && profile->IsOffTheRecord(), &prng)) {
+    accept_language_string += q_values[prng() % q_values.size()];
+  }
+  return accept_language_string;
+}
+
+int OnBeforeStartTransaction_ReduceLanguageWork(
+    net::HttpRequestHeaders* headers,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx) {
+  PrefService* pref_service = user_prefs::UserPrefs::Get(ctx->browser_context);
```

Comment:
no need for user prefs (and the corresponding include), just get them from `profile->GetPrefs()` below